var/spack/repos/spack_repo/builtin/packages/legion/package.py

```diff
@@ -121,10 +121,10 @@ class Legion(CMakePackage, ROCmPackage):
     depends_on("mpi", when="conduit=mpi")
     depends_on("cuda@10.0:11.9", when="+cuda_unsupported_compiler @21.03.0:23.03.0")
     depends_on("cuda@10.0:11.9", when="+cuda @21.03.0:23.03.0")
-    depends_on("cuda@11.7:", when="+cuda_unsupported_compiler @23.06.0:")
-    depends_on("cuda@11.7:", when="+cuda @23.06.0:")
-    depends_on("cuda@11.7:", when="+cuda_unsupported_compiler @stable")
-    depends_on("cuda@11.7:", when="+cuda @stable")
+    depends_on("cuda@11.7:12.8", when="+cuda_unsupported_compiler @23.06.0:25.03.0")
+    depends_on("cuda@11.7:12.8", when="+cuda @23.06.0:25.03.0")
```

Comment:
Please don't add the upper version bounds, because this excludes `master`. We can add the version bound once we make the next Legion release.